include/deal.II/fe/fe_point_evaluation.h

```diff
@@ -48,14 +48,23 @@ namespace internal
       using gradient_type = Tensor<1, n_components, Tensor<1, dim>>;
 
       static void
-      read_value(const double       vector_entry,
+      read_value(const double &     vector_entry,
                  const unsigned int component,
                  value_type &       result)
       {
         AssertIndexRange(component, n_components);
         result[component] = vector_entry;
       }
 
+      static void
+      write_value(double &           vector_entry,
+                  const unsigned int component,
+                  const value_type & result)
+      {
+        AssertIndexRange(component, n_components);
+        vector_entry = result[component];
```

Comment:
I find the names of these functions confusing. The `read_value()` function actually *writes* into `results` argument, whereas the `write_value()` *reads* from the last argument and returns it through the first.

Why not use `set_value()` instead of `read_value()`, and `get_value()` for `write_value()` (possibly returning via a regular `return` rather than through a reference argument)?